code/modules/mob/living/silicon/silicon.dm

```diff
@@ -228,78 +228,92 @@
 /mob/living/silicon/proc/statelaws(force = 0)
 	var/mob/living/silicon/S = usr
 	var/total_laws_count = 0
-	currently_stating_laws = TRUE
-
-	//"radiomod" is inserted before a hardcoded message to change if and how it is handled by an internal radio.
-	say("[radiomod] Current Active Laws:")
-	S.client?.silicon_spam_grace()
 	//laws_sanity_check()
 	//laws.show_laws(world)
 	var/number = 1
-	sleep(10)
 
-	if (laws.devillaws?.len)
+	var/list/laws_to_state = list()
+
+	if (length(laws.devillaws))
 		for(var/index = 1, index <= laws.devillaws.len, index++)
 			if (force || devillawcheck[index] == "Yes")
-				say("[radiomod] 666. [laws.devillaws[index]]")
-				S.client?.silicon_spam_grace()
+				laws_to_state += "666. [laws.devillaws[index]]"
 				total_laws_count++
-				sleep(10)
-
 
 	if (laws.zeroth)
 		if (force || lawcheck[1] == "Yes")
-			say("[radiomod] 0. [laws.zeroth]")
-			S.client?.silicon_spam_grace()
+			laws_to_state += "0. [laws.zeroth]"
 			total_laws_count++
-			sleep(10)
 
 	for (var/index in 1 to laws.hacked.len)
 		var/law = laws.hacked[index]
 		var/num = ion_num()
 		if (length(law) > 0)
 			if (force || hackedcheck[index] == "Yes")
-				say("[radiomod] [num]. [law]")
-				S.client?.silicon_spam_grace()
+				laws_to_state += "[num]. [law]"
 				total_laws_count++
-				sleep(10)
 
 	for (var/index in 1 to laws.ion.len)
 		var/law = laws.ion[index]
 		var/num = ion_num()
 		if (length(law) > 0)
 			if (force || ioncheck[index] == "Yes")
-				say("[radiomod] [num]. [law]")
-				S.client?.silicon_spam_grace()
+				laws_to_state += "[num]. [law]"
 				total_laws_count++
-				sleep(10)
 
 	for (var/index in 1 to laws.inherent.len)
 		var/law = laws.inherent[index]
 
 		if (length(law) > 0)
 			if (force || lawcheck[index+1] == "Yes")
-				say("[radiomod] [number]. [law]")
-				S.client?.silicon_spam_grace()
+				laws_to_state += "[number]. [law]"
 				total_laws_count++
 				number++
-				sleep(10)
 
 	for (var/index in 1 to laws.supplied.len)
 		var/law = laws.supplied[index]
 
 		if (length(law) > 0)
 			if(lawcheck.len >= number+1)
 				if (force || lawcheck[number+1] == "Yes")
-					say("[radiomod] [number]. [law]")
-					S.client?.silicon_spam_grace()
+					laws_to_state += "[number]. [law]"
 					total_laws_count++
 					number++
-					sleep(10)
 
-	S.client?.silicon_spam_grace_done(total_laws_count)
+	var/static/regex/dont_state_regex = regex("Do(?:n'?t| not) state", "i")
```

Comment:
something like `only state laws when there is no human.` should be included

Reply:
This only needs to be simple and not need to catch all cases. This regex matches all recent laws that were uploaded including don't state.

Reply:
I don't think any laws with that wording were in the csv of uploaded AI laws we reviewed in the dev channel anyways?